protocol-units/execution/opt-executor/src/executor/execution.rs

```diff
@@ -93,15 +93,24 @@ impl Executor {
 		let block_executor_clone = block_executor.clone();
 		tokio::task::spawn_blocking(move || {
 			block_executor_clone.commit_blocks(vec![block_id], ledger_info_with_sigs)
-		}).await??;
+		})
+		.await??;
 
 		// commit mempool transactions in batches of size 16
-		for chunk in senders_and_sequence_numbers.chunks(16) {
+		/*for chunk in senders_and_sequence_numbers.chunks(16) {
 			let mut core_mempool = self.core_mempool.write().await;
 			for (sender, sequence_number) in chunk {
+				let _span = debug_span!(
+					target: "movement_timing",
+					"commit_transaction",
+					%sender,
+					sequence_number = *sequence_number,
+				)
+				.entered();
 				core_mempool.commit_transaction(sender, *sequence_number);
+				debug!("committed transaction");
 			}
-		}
+		}*/
```

Comment:
So... if we want to get rid of the mempool lock, this basically needs to happen. 

In a previous usage of the mempool--like way back a couple months ago--this resolved an issue with transaction confirmations. Now, however, it does not seem to be necessary. 

I think this is mainly because we've stopped using the mempool for anything other than checking that a transaction passes the basic mempool validity checks. All of the actual transaction batching and whatnot is just passing through pipe.

Reply:
OK, then the code should be removed.

Do we shift the responsibility on the pre-validator, which is also currently conditionally cut out?